modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/service/ServiceAnalyzerPlugin.java

```diff
@@ -125,6 +150,130 @@ protected void processSpringDependency(Analyzer analyzer) {
 			"com.liferay.portal.spring.extender.service.ServiceReference");
 	}
 
+	private List<String> _buildServiceClasses(Path serviceXml) {
+		List<String> serviceClasses = new ArrayList<>();
+
+		try {
+			DocumentBuilderFactory factory =
+				DocumentBuilderFactory.newInstance();
+
+			DocumentBuilder documentBuilder = factory.newDocumentBuilder();
+
+			Document document = documentBuilder.parse(serviceXml.toFile());
+
+			XPathFactory xPathfactory = XPathFactory.newInstance();
+
+			XPath xpath = xPathfactory.newXPath();
+
+			XPathExpression apiPackagePathExpression = xpath.compile(
+				"/service-builder/@api-package-path");
+
+			String packagePath = apiPackagePathExpression.evaluate(document);
+
+			if ((packagePath == null) || packagePath.isEmpty()) {
+				XPathExpression packagePathExpression = xpath.compile(
+					"/service-builder/@package-path");
+
+				packagePath = packagePathExpression.evaluate(document);
+			}
+
+			if (_isValidPackagePath(packagePath)) {
+				XPathExpression entityExpression = xpath.compile("//entity");
+
+				NodeList entities = (NodeList)entityExpression.evaluate(
+					document, XPathConstants.NODESET);
+
+				for (int i = 0; i < entities.getLength(); i++) {
+					Node entity = entities.item(i);
+
+					String name = _getAttrValue(entity, "name");
+					String localService = _getAttrValue(
+						entity, "local-service");
+					String remoteService = _getAttrValue(
+						entity, "remote-service");
+
+					if ((localService == null) || "true".equals(localService)) {
+						serviceClasses.add(
+							packagePath + ".service." + name + "LocalService");
+					}
+
+					if ("true".equals(remoteService)) {
+						serviceClasses.add(
+							packagePath + ".service." + name + "Service");
+					}
+				}
+			}
+		}
+		catch (Exception e) {
+		}
+
+		return serviceClasses;
+	}
+
+	private Set<Path> _findAllServiceXmlFiles(Path dir) throws IOException {
+		try (Stream<Path> fileTree =
+				Files.walk(dir, FileVisitOption.FOLLOW_LINKS)) {
+
+			return fileTree.filter(
+				path -> {
+					File file = path.toFile();
+
+					String fileName = file.getName();
+
+					return fileName.equals("service.xml");
```

Comment:
This hard codes the name service.xml.  I know that some customers will do things like `service-employees.xml` or `entities.xml` and just change the parameter of service builder gradle task.  My plan for this is to have a followup PR that will add a new bnd plugin-instruction, something like `-servicexml: service-*.xml` so that customers can tell us how to find their service.xml files for when they do things like that.